ui/src/notebooks/context/notebook.tsx

```diff
@@ -55,6 +55,22 @@ export const NotebookProvider: FC = ({children}) => {
     })
   }
 
+  function movePipe(currentIdx: number, newIdx: number) {
+    setPipes(pipes => {
+      const idx = ((newIdx % pipes.length) + pipes.length) % pipes.length
```

Comment:
all of the mods allow us to be lazy with math later on by ensuring the new index is within the bounds of the pipes array